app/models/header_lookup.rb

```diff
@@ -26,70 +22,53 @@ def initialize
   end
 
   def pid_lookup_by_field(term="", field="")
+    searchable_file = ""
+
+    # assign searchable file,  memoized terms, any string modifications
     if term.blank? || field.blank?
-      nil
+      return nil
     elsif field == :mesh
-      @@memoized_mesh[term] || mesh_term_pid_local_lookup(term) || nil
+      searchable_file = @@searchable_mesh_file
+      memoized_terms = @@memoized_mesh
+      memoized_file = MEMOIZED_MESH_FILE
+      term = term.gsub("--", "/")
     elsif field == :lcsh
-      @@memoized_lcsh[term] || lcsh_term_pid_local_lookup(term) || nil
-    elsif field == :subject_name || field == :subject_geographic
-      pid = @@memoized_lcnaf[term] || lcnaf_pid_lookup(term)
-
-      if pid == ABSENT_SUBJECT
-        nil
-      else
-        pid
-      end
+      searchable_file = @@searchable_lcsh_file
+      memoized_terms = @@memoized_lcsh
+      memoized_file = MEMOIZED_LCSH_FILE
+    elsif field == :subject_name || field == :subject_geographic || field = :lcnaf
+      searchable_file = @@searchable_lcnaf_file
+      memoized_terms = @@memoized_lcnaf
+      memoized_file = MEMOIZED_LCNAF_FILE
     else
-      nil
-    end
-  end
-
-  def mesh_term_pid_local_lookup(mesh_term="")
-    File.foreach(@@searchable_mesh_file) do |line|
-      mesh_term_json = JSON.parse(line)
-
-      if mesh_term_json["subject"].downcase == mesh_term.downcase.gsub("--", "/")
-        mesh_id = mesh_term_json["id"]
-        @@memoized_mesh[mesh_term] = mesh_id
-        File.write(MEMOIZED_MESH_FILE, @@memoized_mesh)
-        return mesh_id
-      end
+      return nil
     end
 
-    nil
-  end
+    # check for memoized term
+    if term_pid = memoized_terms[term]
+      term_pid
```

Comment:
minor: having `=` in a conditional is needlessly worrisome
 
```suggestion
    if memoized_terms[term]
      term_pid = memoized_terms[term]
```